linkedList.c

```diff
@@ -0,0 +1,237 @@
+    }
+
+    LinkedListError exit_value = initNode(&(list->head), value);
+
+    if (exit_value != SUCCESS) {
+        return exit_value;
+    }
+
+    list->head->next = tmp;
+
+    return SUCCESS;
+}
+
+/*
+Points the given pointer to the element at the given index of the given list.
+
+@param list - given list.
+@param index - given index.
+@param pointer - given pointer.
+
+@return LinkedListError with exit code.
+*/
+LinkedListError getNodeAt(LinkedList* list, size_t index, Node** pointer) {
+    Node* iterator = list->head;
+
+    if (iterator == NULL) {
+        return INDEX_ERROR;
+    }
+
+    for (int i = 0; i < index; i++) {
+        if (iterator->next == NULL) {
+            return INDEX_ERROR;
+        }
+
+        iterator = iterator->next;
+    }
+
+    *pointer = iterator;
+    return SUCCESS;
+}
+
+LinkedListError insert(LinkedList* list, size_t index, int value) {
+    Node* iterator;
+
+    if (index == 0) {
+        return addFirstNode(list, value);
+    }
+    
+    LinkedListError exit_value = getNodeAt(list, index - 1, &iterator);
+
+    if (exit_value != SUCCESS) {
+        return exit_value;
+    }
+
+    Node* tmp;
+    exit_value = initNode(&tmp, value);
+
+    if (exit_value != SUCCESS) {
+        return exit_value;
+    }
+
+    tmp->next = iterator->next;
+    iterator->next = tmp;
+
+    return SUCCESS;
+}
+
+/*
+Removes the first node of a list.
+
+@param list - list to remove node from.
+
+@return LinkedListError with exit code.
+*/
+LinkedListError removeFirstNode(LinkedList* list) {
+    Node* tmp = list->head->next;
+
+    if (tmp == NULL) {
+        return INDEX_ERROR;
+    }
+
+    free(list->head);
+    list->head = tmp;
+
+    return SUCCESS;
+}
+
+LinkedListError removeAt(LinkedList* list, size_t index) {
+    Node* iterator;
+
+    if (index == 0) {
+        return removeFirstNode(list);
+    }
+
+    LinkedListError exit_value = getNodeAt(list, index - 1, &iterator);
+
+    if (exit_value != SUCCESS) {
+        return exit_value;
+    }
+
+    if (iterator->next == NULL) {
+        return INDEX_ERROR;
+    }
+
+    Node* tmp = iterator->next->next;
+    free(iterator->next);
+    iterator->next = tmp;
+
+    return SUCCESS;
+}
+
+LinkedListError getAt(LinkedList* list, size_t index, int* value) {
+    Node* tmp;
+    LinkedListError exit_value = getNodeAt(list, index, &tmp);
+
+    if (exit_value != SUCCESS) {
+        return exit_value;
+    }
+
+    *value = tmp->value;
+
+    return SUCCESS;
+}
+
+LinkedListError setAt(LinkedList* list, size_t index, int value) {
+    Node* tmp;
+
+    LinkedListError exit_value = getNodeAt(list, index, &tmp);
+
+    if (exit_value != SUCCESS) {
+        return exit_value;
+    }
+
+    tmp->value = value;
+
+    return SUCCESS;
+}
+
+size_t len(LinkedList* list) {
+    size_t size = 0;
+    Node* iterator = list->head;
+
+    while (iterator != NULL) {
+        size++;
+        iterator = iterator->next;
+    }
+
+    return size;
+}
```

Comment:
Can you make this function take `O(1)` time instead of `O(n)`?

What changes does that enforce on the linked list?